modules/spring-config/src/main/java/org/mule/runtime/config/spring/SpringMuleContextServiceConfigurator.java

```diff
@@ -0,0 +1,264 @@
+        beanDefinitionRegistry.registerBeanDefinition(DEFAULT_LOCAL_TRANSIENT_USER_OBJECT_STORE_NAME, getBeanDefinition(DefaultObjectStoreFactoryBean.class, "createDefaultUserTransientObjectStore"));
+        beanDefinitionRegistry.registerBeanDefinition(OBJECT_STORE_MANAGER, getBeanDefinition(MuleObjectStoreManager.class));
+        beanDefinitionRegistry.registerAlias(OBJECT_STORE_MANAGER, LOCAL_OBJECT_STORE_MANAGER);
+    }
+
+    private void createEndpointFactory(BeanDefinitionRegistry beanDefinitionRegistry)
+    {
+        try
+        {
+            Class endpointFactoryClass = ClassUtils.loadClass("org.mule.compatibility.core.endpoint.DefaultEndpointFactory", Thread.currentThread().getContextClassLoader());
+            beanDefinitionRegistry.registerBeanDefinition("_muleEndpointFactory", getBeanDefinition(endpointFactoryClass));
+        }
+        catch (ClassNotFoundException e)
+        {
+            //Nothing to do.
+        }
+    }
+
+    private void createNotificationManager(BeanDefinitionRegistry beanDefinitionRegistry)
+    {
+        List<Notification> defaultNotifications = new ArrayList<>();
+        defaultNotifications.add(new Notification(MuleContextNotificationListener.class, MuleContextNotification.class));
+        defaultNotifications.add(new Notification(SecurityNotificationListener.class, SecurityNotification.class));
+        defaultNotifications.add(new Notification(ManagementNotificationListener.class, ManagementNotification.class));
+        defaultNotifications.add(new Notification(ConnectionNotificationListener.class, ConnectionNotification.class));
+        defaultNotifications.add(new Notification(RegistryNotificationListener.class, RegistryNotification.class));
+        defaultNotifications.add(new Notification(CustomNotificationListener.class, CustomNotification.class));
+        defaultNotifications.add(new Notification(ExceptionNotificationListener.class, ExceptionNotification.class));
+        defaultNotifications.add(new Notification(TransactionNotificationListener.class, TransactionNotification.class));
+        beanDefinitionRegistry.registerBeanDefinition(OBJECT_NOTIFICATION_MANAGER, getBeanDefinitionBuilder(ServerNotificationManagerConfigurator.class)
+                .addPropertyValue("enabledNotifications", defaultNotifications)
+                .getBeanDefinition());
+    }
+
+    private void createBootstrapBeanDefinitions(BeanDefinitionRegistry beanDefinitionRegistry, MuleContext muleContext, ArtifactType artifactType, OptionalObjectsController optionalObjectsController)
+    {
+        try
+        {
+            SpringRegistryBootstrap springRegistryBootstrap = new SpringRegistryBootstrap(artifactType, muleContext, optionalObjectsController, beanDefinitionRegistry);
+            springRegistryBootstrap.initialise();
+        }
+        catch (InitialisationException e)
+        {
+            throw new RuntimeException(e);
```

Comment:
One we can get rid completely of SimpleRegistryBootstrap we will be availabe to remove the InitialisationException thrown by it because the initialise method won't be used anymore in the lifecycle of the MuleContext creation. So it doesn't make sense to throw InitialisationException and force the callers to handle it since we are going to remove it in the future.